test/jdk/java/lang/Class/forName/NonJavaNameTest.java

```diff
@@ -0,0 +1,107 @@
+    private static final String SRC_DIR = System.getProperty("test.src");
+    private static final String NONJAVA_NAMES_SRC = SRC_DIR + "/classes/";
+    private static final String NONJAVA_NAMES_CLASSES = System.getProperty("test.classes", ".");
+    Path dhyphenPath, dcommaPath, dperiodPath, dleftsquarePath, drightsquarePath, dplusPath, dsemicolonPath, dzeroPath, dthreePath, dzadePath;
+
+    @BeforeClass
+    public void createInvalidNameClasses() throws IOException {
+        Path hyphenPath = Paths.get(NONJAVA_NAMES_SRC + "hyphen.class");
+        Path commaPath = Paths.get(NONJAVA_NAMES_SRC + "comma.class");
+        Path periodPath = Paths.get(NONJAVA_NAMES_SRC + "period.class");
+        Path leftsquarePath = Paths.get(NONJAVA_NAMES_SRC + "left-square.class");
+        Path rightsquarePath = Paths.get(NONJAVA_NAMES_SRC + "right-square.class");
+        Path plusPath = Paths.get(NONJAVA_NAMES_SRC + "plus.class");
+        Path semicolonPath = Paths.get(NONJAVA_NAMES_SRC + "semicolon.class");
+        Path zeroPath = Paths.get(NONJAVA_NAMES_SRC + "0.class");
+        Path threePath = Paths.get(NONJAVA_NAMES_SRC + "3.class");
+        Path zadePath = Paths.get(NONJAVA_NAMES_SRC + "Z.class");
+
+        dhyphenPath = Paths.get(NONJAVA_NAMES_CLASSES + "/-.class");
+        dcommaPath = Paths.get(NONJAVA_NAMES_CLASSES + "/,.class");
+        dperiodPath = Paths.get(NONJAVA_NAMES_CLASSES + "/..class");
+        dleftsquarePath = Paths.get(NONJAVA_NAMES_CLASSES + "/[.class");
+        drightsquarePath = Paths.get(NONJAVA_NAMES_CLASSES + "/].class");
+        dplusPath = Paths.get(NONJAVA_NAMES_CLASSES + "/+.class");
+        dsemicolonPath = Paths.get(NONJAVA_NAMES_CLASSES + "/;.class");
+        dzeroPath = Paths.get(NONJAVA_NAMES_CLASSES + "/0.class");
+        dthreePath = Paths.get(NONJAVA_NAMES_CLASSES + "/3.class");
+        dzadePath = Paths.get(NONJAVA_NAMES_CLASSES + "/Z.class");
+
+        Files.copy(hyphenPath, dhyphenPath, REPLACE_EXISTING);
+        Files.copy(commaPath, dcommaPath, REPLACE_EXISTING);
+        Files.copy(periodPath, dperiodPath, REPLACE_EXISTING);
+        Files.copy(leftsquarePath, dleftsquarePath, REPLACE_EXISTING);
+        Files.copy(rightsquarePath, drightsquarePath, REPLACE_EXISTING);
+        Files.copy(plusPath, dplusPath, REPLACE_EXISTING);
+        Files.copy(semicolonPath, dsemicolonPath, REPLACE_EXISTING);
+        Files.copy(zeroPath, dzeroPath, REPLACE_EXISTING);
+        Files.copy(threePath, dthreePath, REPLACE_EXISTING);
+        Files.copy(zadePath, dzadePath, REPLACE_EXISTING);
+    }
+
+    @Test
+    public void NonJavaNamestest() throws Throwable {
+        ProcessBuilder processBuilder = ProcessTools.createJavaProcessBuilder( "NonJavaNames");
+        OutputAnalyzer outputAnalyzer = ProcessTools.executeCommand(processBuilder);
+        outputAnalyzer.shouldHaveExitValue(0);
+    }
+
+    @AfterClass
+    public void deleteInvalidNameClasses() throws IOException {
```

Comment:
jtreg will do the clean up and this is not necessary.

Reply:
Removed in next patch.